api.planx.uk/modules/analytics/metabase/collection/types.ts

```diff
@@ -0,0 +1,48 @@
+import { z } from "zod";
+import type { ValidatedRequestHandler } from "../../../../shared/middleware/validate.js";
+import type { ApiResponse } from "../shared/types.js";
+
+/** Interface for incoming request, in camelCase */
+export interface NewCollectionParams {
+  slug: string;
+  description?: string;
+  /** Optional; if the collection is a child of a parent, specify parent ID here */
+  parentId?: number;
+}
+
+/** Interface for request after transforming to snake case (Metabase takes snake while PlanX API takes camel) */
+export type MetabaseCollectionParams = Omit<NewCollectionParams, "slug"> & {
+  name: string;
+};
+
+/** TODO: when running on production, turn below comment back into code
+ * the Metabase collection ID is for the "Council" collection
+ * see https://github.com/theopensystemslab/planx-new/pull/4072#discussion_r1892631692
+ **/
+// const COUNCILS_COLLECTION_ID = 58;
+
+export const createTeamCollectionSchema = z.object({
+  params: z.object({
+    slug: z.string(),
```

Comment:
Just seen the "long slug" test - if Metabase has a restriction here (presumably 100 based on test?), we can also catch this early here. 

```suggestion
    slug: z.string().max(100),
```

Reply:
There's no restriction on PlanX team names so it's _possible_ that an team name could exceed 100 characters (Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch 😉 ).

If this were to ever happen I'm sure we can limit and make unique in some way 👍

Reply:
Don't think Metabase actually has a limit, just thought it was something that might be worth testing / went with an arbitrary number, so will leave out of validation for now!